src/objects/Mesh.js

```diff
@@ -160,9 +160,9 @@ THREE.Mesh.prototype.raycast = ( function () {
 							distance: distance,
 							point: intersectionPoint,
 							face: new THREE.Face3( a, b, c, THREE.Triangle.normal( vA, vB, vC ) ),
-							faceIndex: null,
+							faceIndex: Math.floor(i/3), // triangle number in indices buffer semantics
```

Comment:
This is not sufficient information, correct? You need to return drawcall (offset) `oi` and index `i` at a minimum. How did you test this?